### DISCUSSION TITLE
spec.go vs spec_go111.go - swagger generate spec command

### DISCUSSION THREAD

**Discussion by Huckletoon** [go-swagger/go-swagger#2711] [Q&A] [answered]
Hey all, I have a couple of questions I'm hoping to get some insight on.

1. I've done some digging through the repo and I'm at a little bit of a loss regarding the `swagger generate spec` command. When building the binary, what determines whether `cmd/swagger/generate/spec.go` or `cmd/swagger/generate/spec_go111.go` is picked and used for this command? In building locally both inside and outside of Docker, it seems that `spec_go111.go` is used, but I'm not sure why (it might be something with my local environment but again, I'm not sure).

2. Another thing that might clear up my confusion surrounding the first question - what exactly is the difference and purpose of these two different spec command sources? (E.g. what am I missing? 😅)

Any help, answers, and insight would be greatly appreciated - thank you in advance!

**Comment by casualjim** [comment, marked as answer]
These files exist since there was a change in go 1.11 that changed the way we had to walk the code at compile time. So those 2 files exist for backwards compatibility. I think by now it's safe to remove support for go versions < 1.11 

The mechanism for picking the files to include at compile time are [build constraints](https://pkg.go.dev/cmd/go@go1.16.13#hdr-Build_constraints). 

The files that are using the new way after go1.11 include the following [build constraint](https://github.com/go-swagger/go-swagger/blob/master/cmd/swagger/commands/generate/spec_go111.go#L1):

```go
// +build go1.11
````

These files use the [codescan](https://github.com/go-swagger/go-swagger/blob/master/cmd/swagger/commands/generate/spec_go111.go#L12) package

And the other files that include the following [build constraint](https://github.com/go-swagger/go-swagger/blob/master/cmd/swagger/commands/generate/spec.go#L1):

```go
// +build !go1.11
```

These files use the [scan](https://github.com/go-swagger/go-swagger/blob/master/cmd/swagger/commands/generate/spec.go#L28) package.
